src/js/controllers/index.js

```diff
@@ -782,6 +735,114 @@ angular.module('copayApp.controllers').controller('indexController', function($r
     });
   };
 
+  self.stopSync = function(remoteTx, localTx) {
+    if (remoteTx.txid == localTx.txid)
+      return true;
+    else
+      return false;
+  }
+
+  self.removeLessThanSixConfirmations = function(txs) {
+    return lodash.map(txs, function(tx) {
+      if (tx.confirmations >= CONFIRMATIONS)
+        return tx;
+    });
+  }
+
+  self.updateLocalTxHistory = function(cb) {
+    var fc = profileService.focusedClient;
+    var c = fc.credentials;
+    var txsToPush = [];
+
+    storageService.getTxHistory(c.walletId, function(err, txs) {
+      if (err) return cb(err);
+
+      var localTxs;
+      try {
+        localTxs = JSON.parse(txs);
+      } catch (ex) {
+        return cb(ex);
+      }
+
+      if (!localTxs)
+        localTxs = [];
+
+      var txsFromLocal = self.removeLessThanSixConfirmations(localTxs);
+
+      var count = 0;
+      fillTxsObject(txsToPush);
+
+      function fillTxsObject(txsToPush) {
+        self.makeTxHistoryRequest(txsToPush, txsFromLocal, function(err, exitLoop, txsResult) {
+          if (err) return cb(err);
+          if (exitLoop) {
+            self.txHistory = [];
+            self.setTxHistory(lodash.compact(txsResult.concat(txsFromLocal)));
+            storageService.setTxHistory(JSON.stringify(self.txHistory), c.walletId, function() {
+              return cb(null);
+            });
+          } else
+            fillTxsObject(txsResult);
+        });
+      };
+    });
+  }
+
+  self.makeTxHistoryRequest = function(txsToPush, localTx, cb) {
```

Comment:
instead of passing `localTx[]`, it should use `lastTx` (only one TX, not an array).